pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -0,0 +1,130 @@
+package dynamic
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	sdkauthorization "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armauthorization"
+	"github.com/Azure/ARO-RP/pkg/util/rbac"
+	"github.com/Azure/ARO-RP/pkg/util/version"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (dv *dynamic) ValidatePlatformWorkloadIdentityProfile(ctx context.Context, oc *api.OpenShiftCluster, platformWorkloadIdentityRolesByRoleName map[string]api.PlatformWorkloadIdentityRole, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	dv.log.Print("ValidatePlatformWorkloadIdentityProfile")
+
+	dv.platformIdentitiesActionsMap = map[string][]string{}
+	dv.platformIdentities = []api.PlatformWorkloadIdentity{}
+
+	for _, pwi := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+		_, ok := platformWorkloadIdentityRolesByRoleName[pwi.OperatorName]
+		if ok {
+			dv.platformIdentitiesActionsMap[pwi.OperatorName] = nil
+			dv.platformIdentities = append(dv.platformIdentities, pwi)
+		}
+	}
+
+	// Check if any required platform identity is missing
+	if len(dv.platformIdentities) != len(platformWorkloadIdentityRolesByRoleName) {
+		requiredOperatorIdentities := []string{}
+		for _, role := range platformWorkloadIdentityRolesByRoleName {
+			requiredOperatorIdentities = append(requiredOperatorIdentities, role.OperatorName)
+		}
+		v := oc.Properties.ClusterProfile.Version
+		currentOpenShiftVersion, err := version.ParseVersion(oc.Properties.ClusterProfile.Version)
+		if err != nil {
+			return err
+		}
+		if oc.Properties.PlatformWorkloadIdentityProfile.UpgradeableTo != nil {
+			upgradeableVersion, err := version.ParseVersion(string(*oc.Properties.PlatformWorkloadIdentityProfile.UpgradeableTo))
+			if err != nil {
+				return err
+			}
+			if !upgradeableVersion.Eq(currentOpenShiftVersion) && currentOpenShiftVersion.Lt(upgradeableVersion) {
```

Comment:
This can be simplified; if we know that the current version is less than the upgrade version, we know they aren't equal:

```suggestion
			if currentOpenShiftVersion.Lt(upgradeableVersion) {
```

Reply:
So the check here is to make sure that the Minor version is also not same. For eg, if your suggested change is performed, it will be recurring to consider upgradeableVersion for validation upgradeableVersion = 4.14.60 and currentVersion = 4.14.50 .
Here the equality check is incorrect, it should be done between the minor versions. Performing the change here.

Reply:
Done, updated the comparison and also added unit test cases for it.